python/paddle/fluid/contrib/int8_inference/utility.py

```diff
@@ -153,143 +163,93 @@ def __get_max_range_by_var_name(self, program, var_name):
-            if not i in Calibrator.supported_int8_op_type:
-                return False
-        return True
-
-    def __check_var_source_dt(self, var_name):
-        '''
-        Check whether the specified variable is the output of int8 conv op or not.
-        If true, return the original op index.
-        If false, return -1
-        '''
-        return self._int8_output_var_op_index_dict[
-            var_name] if var_name in self._int8_output_var_op_index_dict else -1
-
-    def __update_int8_output_var_op_index_dict(self, index, var_name=None):
-        '''
-        Update the int8_output_variable/op_index dictionary
-        '''
-        for k, v in self._int8_output_var_op_index_dict.items():
-            if v >= index:
-                self._int8_output_var_op_index_dict[k] = v + 1
-        if var_name:
-            self._int8_output_var_op_index_dict[var_name] = index
-
     def __update_program(self):
         '''
         Update the program with the quantize/dequantize op insertion.
         '''
-        quantize_index, dequantize_index = self.__get_quantize_dequantize_combination(
+        quantize_indexes, dequantize_indexes, next_diff_ops_map = self.__get_quantize_dequantize_combination(
             self._output_program)
-        inserted_op_length = 0
         calc_max_func = self.__get_optimal_scaling_factor if self.algo == "KL" else np.max
-        insert_op_collection = sorted(quantize_index + dequantize_index)
-
-        for index in insert_op_collection:
-            if index in quantize_index:
-                quantize_tmp = self._output_program.current_block().create_var(
-                    name="quantize_{}_tmp".format(index),
-                    dtype=core.VarDesc.VarType.UINT8)
-                original_out_name = self._output_program.current_block().ops[
-                    index + inserted_op_length - 1].output_names[0]
-                original_out = self._output_program.current_block().ops[
-                    index + inserted_op_length - 1].output(original_out_name)[0]
-
+        insert_poses = quantize_indexes + dequantize_indexes
+        insert_poses.sort(reverse=True)
+        quantize_index = len(quantize_indexes) - 1
+        dequantize_index = len(dequantize_indexes) - 1
+        for i in insert_poses:
+            if i == -1 and self.first_conv_int8:
+                current_out = self._output_program.global_block().ops[
+                    self._conv_op_index[0]].input_arg_names[2]
+            else:
+                current_op = self._output_program.global_block().ops[i]
+                current_out = current_op.output_arg_names[0]
+                if current_op.type == "split":
+                    current_out = current_op.output_arg_names[1]
+            if i in quantize_indexes:
+                var_tmp = self._output_program.current_block().create_var(
+                    name="quantize_{}_tmp".format(quantize_index),
+                    dtype=core.VarDesc.VarType.UINT8, )
```

Comment:
Modified

Reply:
this one is not modified I think?

Reply:
I took out the op connection part from the if-else. Now it only keep the part that creates the variable and quantize/dequantize op. I think it can't be simplified anymore.  How do you think?

Reply:
Ok sorry. its fine